### PR TITLE
CONSOLE-3061: Add failed plugins into the notification drawer

### PR DESCRIPTION
**Pull request by andrewballantyne**
https://issues.redhat.com/browse/CONSOLE-3061

Adds support for failed plugins to appear in the notification drawer.
Screen Shot 2022-06-20 at 2 29 48 PM

I took some liberties with what's inside the notification (omitted from the UX design), but the rest is from the ticket screenshots. (NOTE: I ran this by Thi -- not on GitHub -- and she approved the text)

/assign @rhamilto

### CODE REVIEW

File: frontend/packages/console-plugin-sdk/src/store.ts [hunk 1]
```diff
@@ -306,3 +306,7 @@ export type DynamicPluginInfo = LoadedDynamicPluginInfo | NotLoadedDynamicPlugin
 
 export const isLoadedDynamicPluginInfo = (i: DynamicPluginInfo): i is LoadedDynamicPluginInfo =>
   i.status === 'Loaded';
+
+export const isNotLoadedDynamicPluginInfo = (
+  i: DynamicPluginInfo,
+): i is NotLoadedDynamicPluginInfo => i.status === 'Failed' || i.status === 'Pending';
```
Comment (andrewballantyne) [inline frontend/packages/console-plugin-sdk/src/store.ts:312]: This mirrors the code in @rhamilto's PR - https://github.com/openshift/console/pull/11664

File: frontend/public/components/notification-drawer.tsx [hunk 2]
```diff
@@ -179,6 +188,27 @@ const getUpdateNotificationEntries = (
...
+  if (failedPlugins.length > 0) {
+    entries.push(
+      ...failedPlugins.map((plugin) => {
+        const link = resourcePath(referenceForModel(ConsolePluginModel), plugin.pluginName);
+        return (
+          <NotificationEntry
+            actionPath={link}
+            actionText={i18next.t('public~View plugin')}
+            key={`${plugin.pluginName}-dynamic-plugin-fail`}
+            description={i18next.t('public~Something went wrong with the {{pluginName}} plugin.', {
```
Comment (spadgett) [inline frontend/public/components/notification-drawer.tsx:200]: OK as a follow up, but I think we should try to capture the original error to help admins debug. For instance, if there is an HTTP error loading the plugin manifest, we can add that to the status in the plugin store and show it here.

cc @vojtechszocs

Comment (andrewballantyne) [reply frontend/public/components/notification-drawer.tsx:200]: This should mirror whatever in the dashboard status popup too

**Comment by RickJWagner** [comment]
<Ack> Thanks for the explanation of what's provided.  
Hopefully there will be a finite number of ways a plugin can fail, when CEE gathers a list of these over time (in a KCS) even the hint found in the Notification Drawer might be enough of a clue to fix things.

**Comment by yapei** [comment]
verified that failed loading plugins will appear in notification drawer, also a link is provided to view plugin details
Screen Shot 2022-06-22 at 11 32 15
/label qe-approved

**Comment by andrewballantyne** [comment]
cc @opayne1 If you could get a moment to review & approve (or ask questions) before EoD that would help with Code Freeze tomorrow.

**Pull request merged**